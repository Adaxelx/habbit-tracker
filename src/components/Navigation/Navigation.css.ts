import styled from 'styled-components';
// import { getKey } from 'utils/objectsFunctions';

// const topLeft = css`
//   top: 0;
//   left: 0;
// `;

// const topRight = css`
//   top: 0;
//   right: 0;
// `;

// const bottomLeft = css`
//   bottom: 0;
//   left: 0;
// `;

// const bottomRight = css`
//   bottom: 0;
//   right: 0;
// `;

// interface Position {
//   topLeft: FlattenSimpleInterpolation;
//   topRight: FlattenSimpleInterpolation;
//   bottomLeft: FlattenSimpleInterpolation;
//   bottomRight: FlattenSimpleInterpolation;
//   [key: string]: FlattenSimpleInterpolation;
// }

// const positionObject: Position = { topLeft, topRight, bottomLeft, bottomRight };

// interface DotProps {
//   topLeft?: boolean;
//   topRight?: boolean;
//   bottomLeft?: boolean;
//   bottomRight?: boolean;
//   [key: string]: boolean | undefined;
// }

/* background-color: ${({ theme, ...rest }) => theme.colors.nav[getKey(rest)]};
  ${({ theme, ...rest }) => positionObject[getKey(rest)]} */

interface ClicableProps {
  open: boolean;
}

export const StyledNavButton = styled.button<ClicableProps>`
  position: fixed;
  height: 50px;
  width: 50px;
  top: ${({ theme }) => theme.margin.sm};
  background-color: transparent;
  border: none;
  right: ${({ theme }) => theme.margin.sm};
  padding: 0;
  ${({ open }) => open && `transform: rotate(-45deg);`}
  transition: ${({ theme }) => `${theme.time.medium}ms`};
  z-index: ${({ theme }) => theme.zIndex.max};
  &:focus {
    outline: none;
  }
`;

const StyledDot = styled.div<ClicableProps>`
  width: 20px;
  height: 20px;
  border-radius: 50%;
  position: absolute;
  opacity: ${({ open }) => (open ? '0' : '1')};
  transition: ${({ theme }) => `${theme.time.medium}ms`};
`;

export const StyledTLDot = styled(StyledDot)`
  background-color: ${({ theme }) => theme.colors.nav.topLeft};
  top: 0;
  left: 0;
`;

export const StyledTRDot = styled(StyledDot)`
  background-color: ${({ theme }) => theme.colors.nav.topRight};
  top: 0;
  right: 0;
`;

export const StyledBLDot = styled(StyledDot)`
  background-color: ${({ theme }) => theme.colors.nav.bottomLeft};
  bottom: 0;
  left: 0;
`;

export const StyledBRDot = styled(StyledDot)`
  background-color: ${({ theme }) => theme.colors.nav.bottomRight};
  bottom: 0;
  right: 0;
`;

export const StyledLine = styled.div<ClicableProps>`
  position: absolute;
  height: 3px;
  width: 50px;
  background-color: ${({ theme }) => theme.colors.error};
  opacity: ${({ open }) => (open ? '1' : '0')};
  transition: ${({ theme }) => `${theme.time.medium}ms`};
`;

export const StyledRotatedLine = styled(StyledLine)<ClicableProps>`
  transform: rotate(90deg);
  top: 50%;
`;
