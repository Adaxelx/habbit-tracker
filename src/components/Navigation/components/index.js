export { default as OpenedNavigation } from './OpenedNavigation';
