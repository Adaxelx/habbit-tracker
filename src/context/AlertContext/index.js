export { default, useAlertContext } from './AlertContext';
