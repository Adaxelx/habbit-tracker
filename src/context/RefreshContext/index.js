export { default, useRefreshContext } from './RefreshContext';
