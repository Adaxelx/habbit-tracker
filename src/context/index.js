export { default as ThemeProvider, useThemeContext } from './ThemeContext';
export { default as UserProvider, useUserContext, UserContext } from './UserContext';
export { default as AlertProvider, useAlertContext, AlertContext } from './AlertContext';
export { default as RefreshProvider, useRefreshContext } from './RefreshContext';
