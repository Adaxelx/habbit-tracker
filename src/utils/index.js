export { default as TestUtil } from './TestUtil';
export { default as RenderWithRouter } from './RenderWithRouter';
