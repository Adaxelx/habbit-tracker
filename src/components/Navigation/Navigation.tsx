import React, { useState } from 'react';
import {
  StyledNavButton,
  StyledTLDot,
  StyledTRDot,
  StyledBLDot,
  StyledBRDot,
  StyledLine,
  StyledRotatedLine,
} from './Navigation.css';
import { OpenedNavigation } from './components';

const Navigation = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <StyledNavButton open={open} onClick={() => setOpen((prevState) => !prevState)}>
        <StyledTLDot open={open} />
        <StyledTRDot open={open} />
        <StyledBLDot open={open} />
        <StyledBRDot open={open} />
        <StyledLine open={open} />
        <StyledRotatedLine open={open} />
      </StyledNavButton>
      <OpenedNavigation open={open} handleClose={() => setOpen(false)} />
    </>
  );
};

export default Navigation;
