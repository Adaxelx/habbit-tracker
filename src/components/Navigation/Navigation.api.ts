import APIpaths from 'constants/APIpaths';
import { TokenType } from 'utils';

export const logoutUser = async (token: TokenType) => {
  const response = await fetch(APIpaths.LOGOUT, {
    headers: { 'Content-Type': 'application/json', Authorization: `${token}` },
    method: 'POST',
  });
  if (response.status === 200) {
    return response.json();
  }
  const { message } = await response.json();
  throw new Error(message || 'Something went wrong');
};
