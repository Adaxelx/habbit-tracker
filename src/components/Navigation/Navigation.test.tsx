import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'context';
import { BrowserRouter } from 'react-router-dom';
import { RenderWithRouter } from 'utils';
import Navigation from './Navigation';

describe('Navigation', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should open nav and redirect after click', async () => {
    const { container, getByTestId, getByText } = RenderWithRouter(
      <ThemeProvider>
        <BrowserRouter>
          <Navigation />
        </BrowserRouter>
      </ThemeProvider>,
    );

    fireEvent.click(getByTestId('navButton'));

    expect(container).toMatchSnapshot();

    expect(window.location.pathname).toBe('/');

    fireEvent.click(getByText('Login'));

    expect(window.location.pathname).toBe('/login');
  });
});
