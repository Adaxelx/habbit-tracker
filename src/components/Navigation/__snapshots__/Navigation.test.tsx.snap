// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
<div>
  <button
    class="sc-bdfBwQ sdFZx"
    data-testid="navButton"
  >
    <div
      class="sc-gsTCUz sc-dlfnbm bytOLq iNoqna"
    />
    <div
      class="sc-gsTCUz sc-hKgILt bytOLq dZJXyp"
    />
    <div
      class="sc-gsTCUz sc-eCssSg bytOLq eLeZFd"
    />
    <div
      class="sc-gsTCUz sc-jSgupP bytOLq imevjb"
    />
    <div
      class="sc-gKsewC wEEon"
    />
    <div
      class="sc-gKsewC sc-iBPRYJ wEEon ekDqeo"
    />
  </button>
  <nav
    class="sc-fodVxV iYzKGD"
  >
    <div
      class="sc-fFubgz fTHQZi"
    >
      <a
        class="sc-bkzZxe ddUFlK"
        href="/"
      >
        Home page
      </a>
      <a
        class="sc-bkzZxe ddUFlK"
        href="/calendar"
      >
        Your tracker
      </a>
      <a
        class="sc-bkzZxe ddUFlK"
        href="/login"
      >
        Login
      </a>
      <a
        class="sc-bkzZxe ddUFlK"
        href="/register"
      >
        Register
      </a>
    </div>
  </nav>
</div>
`;

exports[`Navigation should open nav and redirect after click 1`] = `
<div>
  <button
    class="sc-bdfBwQ kMWbBq"
    data-testid="navButton"
    open=""
  >
    <div
      class="sc-gsTCUz sc-dlfnbm fHaXLb iNoqna"
      open=""
    />
    <div
      class="sc-gsTCUz sc-hKgILt fHaXLb dZJXyp"
      open=""
    />
    <div
      class="sc-gsTCUz sc-eCssSg fHaXLb eLeZFd"
      open=""
    />
    <div
      class="sc-gsTCUz sc-jSgupP fHaXLb imevjb"
      open=""
    />
    <div
      class="sc-gKsewC JhcfG"
      open=""
    />
    <div
      class="sc-gKsewC sc-iBPRYJ JhcfG ekDqeo"
      open=""
    />
  </button>
  <nav
    class="sc-fodVxV eLNwrj"
    open=""
  >
    <div
      class="sc-fFubgz fTHQZi"
    >
      <a
        class="sc-bkzZxe kEFHMp"
        href="/"
        open=""
      >
        Home page
      </a>
      <a
        class="sc-bkzZxe kEFHMp"
        href="/calendar"
        open=""
      >
        Your tracker
      </a>
      <a
        class="sc-bkzZxe kEFHMp"
        href="/login"
        open=""
      >
        Login
      </a>
      <a
        class="sc-bkzZxe kEFHMp"
        href="/register"
        open=""
      >
        Register
      </a>
    </div>
  </nav>
</div>
`;
